third_party/move/move-compiler-v2/src/file_format_generator/module_generator.rs

```diff
@@ -75,8 +75,8 @@ pub struct ModuleGenerator {
         BTreeMap<(QualifiedId<StructId>, usize, FF::SignatureIndex), FF::FieldInstantiationIndex>,
     /// A mapping from type sequences to signature indices.
     types_to_signature: BTreeMap<Vec<Type>, FF::SignatureIndex>,
-    /// A mapping from constants sequences to pool indices.
-    cons_to_idx: BTreeMap<Constant, FF::ConstantPoolIndex>,
+    /// A mapping from constants sequences (with the corresponding type information) to pool indices.
+    cons_to_idx: BTreeMap<(Constant, Type), FF::ConstantPoolIndex>,
```

Comment:
Use (constant, type) so that constants having the same value but with different types have different constant pool index.